comment_scan/commander.rb

```diff
@@ -104,338 +105,339 @@ def to_sizes(filenames)
 
     #TODO: consider making some kind of check valid arg that'll return a bool and print "Bad! Arg must be in {X, Y, Z}"
     #      since this is definitely something I do several times below
-end
 
-def whoami(commander, room_id)
-    return unless commander.on?(room_id)
+     def whoami(room_id)
+        return unless on?(room_id)
 
-    if commander.isHQ?(room_id) && rand(0...20) == rand(0...20)
-        commander.chatter.say("24601")
-    else
-        commander.chatter.say("I go by #{commander.BOT_NAMES.join(" and ")}", room_id)
+        if isHQ?(room_id) && rand(0...20) == rand(0...20)
+            @chatter.say("24601")
+        else
+            @chatter.say("I go by #{@BOT_NAMES.join(" and ")}", room_id)
+        end
     end
-end
-
-def alive(commander, room_id, bot='*')
-    return unless commander.matches_bot?(bot) || !commander.on?(room_id)
-    commander.chatter.say("I'm alive and well :)", room_id)
-end
 
-def on(commander, room_id, bot='*')
-    return unless commander.matches_bot?(bot)
-    if Room.on?(room_id)
-        commander.chatter.say("I'm already on, silly", room_id)
-    else
-        commander.chatter.say("Turning on...", room_id)
-        Room.turn_on(room_id)
+    def alive(room_id, bot='*')
+        return unless matches_bot?(bot) || !on?(room_id)
+        @chatter.say("I'm alive and well :)", room_id)
     end
-end
 
-def off(commander, room_id, bot='*')
-    return unless commander.matches_bot?(bot)
-    if !Room.on?(room_id)
-        commander.chatter.say("I'm already off, silly", room_id)
-    else
-        commander.chatter.say("Turning off...", room_id)
-        Room.turn_off(room_id)
+    def on(room_id, bot='*')
+        return unless matches_bot?(bot)
```

Comment:
In theory, for all these `return unless matches_bot?(bot)` we could have some fun and abuse exceptions to make it a bit neater, but that's not terribly important.

Reply:
Ooh yea that was something that bugged me, but I couldn't think of a nice way to get rid of them. Even with exceptions, we'd still have a line at the top of every function, no? I was thinking of passing some flag for whether or not to check bot_name that I'd pass to chatter. But I'm not sure if I'd want that logic in there.